test/Microsoft.DotNet.ShellShim.Tests/OsxEnvironmentPathTests.cs

```diff
@@ -2,82 +2,152 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
-using System.Collections.Generic;
-using System.Diagnostics;
-using System.Runtime.InteropServices;
 using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
 using Microsoft.DotNet.Configurer;
 using Microsoft.DotNet.Tools;
 using Microsoft.DotNet.Tools.Test.Utilities;
 using Microsoft.Extensions.DependencyModel.Tests;
+using Moq;
 using Xunit;
 
 namespace Microsoft.DotNet.ShellShim.Tests
 {
     public class OsxEnvironmentPathTests
     {
         [Fact]
-        public void GivenEnvironmentAndReporterItCanPrintOutInstructionToAddPath()
+        public void GivenPathNotSetItPrintsManualInstructions()
         {
             var reporter = new BufferedReporter();
-            var osxEnvironmentPath = new OSXEnvironmentPath(
-                new BashPathUnderHomeDirectory("/myhome", "executable/path"),
+            var toolsPath = new BashPathUnderHomeDirectory("/home/user", ".dotnet/tools");
+            var pathValue = @"/usr/bin";
+            var provider = new Mock<IEnvironmentProvider>(MockBehavior.Strict);
+
+            provider
+                .Setup(p => p.GetEnvironmentVariable("PATH"))
+                .Returns(pathValue);
+
+            var environmentPath = new OSXEnvironmentPath(
+                toolsPath,
                 reporter,
-                new FakeEnvironmentProvider(
-                    new Dictionary<string, string>
-                    {
-                        {"PATH", ""}
-                    }),
-                FakeFile.Empty);
+                provider.Object,
+                FileSystemMockBuilder.Empty.File);
 
-            osxEnvironmentPath.PrintAddPathInstructionIfPathDoesNotExist();
+            environmentPath.PrintAddPathInstructionIfPathDoesNotExist();
 
-            // similar to https://code.visualstudio.com/docs/setup/mac
             reporter.Lines.Should().Equal(
                 string.Format(
                     CommonLocalizableStrings.EnvironmentPathOSXManualInstructions,
-                    "/myhome/executable/path"));
+                    toolsPath.Path));
+        }
+
+        [Fact]
+        public void GivenPathNotSetAndProfileExistsItPrintsReopenMessage()
```

Comment:
I have very low confident on tests around env:path. Since we depend on shell's behavior. And all we can do is set this set that. How, it is being considered is on the specific shell.

So, whatever you touch. You need to run it manually. However, seems like only outside visible change is windows first run. You just need to very it by hand.

Also, please make sure it won't happen on first run again and again. First run have at least 3 flags. disable first run, disable telemetry, and actual nuget cache existence(?).

Reply:
I'll verify that disabling first run experience and the sentinels prevent us from attempting to add the path again.  I don't want to hit this exception in this environment every time dotnet starts either.

Reply:
So you're right that `DOTNET_SKIP_FIRST_TIME_EXPERIENCE` will be ignored for the purpose of skipping adding the path to the environment.  I will fix that.

However, `DOTNET_PRINT_TELEMETRY_MESSAGE` is tied to the creation of the first run notice sentinel, which we're using with the tool path as well :/

User doesn't set `DOTNET_SKIP_FIRST_TIME_EXPERIENCE` but sets `DOTNET_PRINT_TELEMETRY_MESSAGE` to false and tools path is attempted every time.

Reply:
I think that is the long blocker for this PR

Reply:
@livarcocc This was the issue I mentioned in stand-up.  Should we use yet another sentinel to track this or live with users setting `DOTNET_PRINT_TELEMETRY_MESSAGE` but not `DOTNET_SKIP_FIRST_TIME_EXPERIENCE` always attempting to modify the path (only happens if the expected directory isn't already on the path, though).

Reply:
`DOTNET_SKIP_FIRST_TIME_EXPERIENCE` should turn this off.
`DOTNET_PRINT_TELEMETRY_MESSAGE` should not.

Reply:
Ok then we need either another sentinel or we need to set the first-run notice sentinel (which we're using right now for this) despite `DOTNET_PRINT_TELEMETRY_MESSAGE` being set.